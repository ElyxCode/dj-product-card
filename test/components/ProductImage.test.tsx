import React from 'react';
import renderer from 'react-test-renderer';
import { ProductImage, ProductCard } from '../../src/components';
import { product2 } from '../data/product';

describe('ProductImage', () => {
  test('should display the componente correctly with url img ', () => {
    const wrapper = renderer.create(<ProductImage img="./coffee-img.jpg" />);

    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  test('should display the component with product iamge', () => {
    const wrapper = renderer.create(
      <ProductCard product={product2}>{() => <ProductImage />}</ProductCard>
    );

    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
