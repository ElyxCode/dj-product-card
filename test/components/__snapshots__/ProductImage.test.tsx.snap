// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage should display the component with product iamge 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Product image"
    className="productImg undefined"
    src="./coffee-mug.png"
  />
</div>
`;

exports[`ProductImage should display the componente correctly with url img  1`] = `
<img
  alt="Product image"
  className="productImg undefined"
  src="./coffee-img.jpg"
/>
`;
